packages/fiori/src/SearchField.ts

```diff
@@ -0,0 +1,275 @@
+ * @public
+ */
+@event("search", {
+	bubbles: true,
+})
+
+class SearchField extends UI5Element {
+	eventDetails!: {
+		search: void,
+		input: void,
+		"scope-change": SearchFieldScopeSelectionChangeDetails,
+	}
+
+	/**
+	 * Defines the mode of the component.
+	 * @default "Default"
+	 * @public
+	 */
+	@property()
+	mode: `${SearchMode}` = "Default";
+
+	/**
+	 * Defines whether the clear icon of the search will be shown.
+	 * @default false
+	 * @public
+	 */
+	@property({ type: Boolean })
+	showClearIcon = false;
+
+	/**
+	 * Defines whether the component is expanded.
+	 *
+	 * @default false
+	 * @public
+	 */
+	@property({ type: Boolean })
+	expanded = false;
+
+	/**
+	 * Determines whether the component is in a fixed state that is not
+	 * expandable/collapsible by user interaction.
+	 * @default false
+	 * @public
+	 */
+	@property({ type: Boolean })
+	fixed = false;
+
+	/**
+	 * Defines the value of the component.
+	 *
+	 * **Note:** The property is updated upon typing.
+	 * @default ""
+	 * @public
+	 */
+	@property()
+	value = "";
+
+	/**
+	 * Defines a short hint intended to aid the user with data entry when the
+	 * component has no value.
+	 * @default undefined
+	 * @public
+	 */
+	@property()
+	placeholder?: string;
+
+	/**
+	 * Defines the accessible ARIA name of the component.
+	 * @public
+	 * @default undefined
+	 */
+	@property()
+	accessibleName?: string;
+
+	/**
+	 * Defines the component scope options.
+	 * @public
+	 */
+	@slot({ type: HTMLElement, individualSlots: true, invalidateOnChildChange: true })
+	scopeOptions!: Array<ISearchFieldScopeOption>;
+
+	/**
+	 * @private
+	 */
+	@property({ type: Boolean })
+	focusedInnerInput = false;
+
+	/**
+	 * @private
+	 */
+	@property({ type: Boolean })
+	_effectiveShowClearIcon = false;
+
+	@i18n("@ui5/webcomponents-fiori")
+	static i18nBundle: I18nBundle;
+
+	onBeforeRendering() {
+		this._effectiveShowClearIcon = (this.showClearIcon && !!this.value);
+	}
+
+	_onkeydown(e:KeyboardEvent) {
+		if (isEnter(e)) {
+			return this._handleEnter();
+		}
+	}
+
+	_onfocusin() {
+		this.focusedInnerInput = true;
+	}
+
+	_onfocusout() {
+		this.focusedInnerInput = false;
+	}
+
+	_handleEnter() {
+		if (this.value.length) {
+			this.fireDecoratorEvent("search");
+		}
+	}
+
+	_handleSearchIconFocusOut() {}
+
+	_handleSearchIconPress() {
+		if (this.value.length) {
+			this.fireDecoratorEvent("search");
+			return;
+		}
+
+		if (this.fixed) {
+			return;
+		}
+
+		this.expanded = !this.expanded;
+
+		setTimeout(() => {
+			this.focus();
+		}, 0);
+	}
+
+	_handleInput(e: InputEvent) {
+		this.value = (e.target as HTMLInputElement).value;
+
+		this.fireDecoratorEvent("input");
+	}
+
+	_handleClear() {
+		this.value = "";
+		this.fireDecoratorEvent("input");
+
+		this.focus();
+	}
+
+	_handleScopeChange(e: UI5CustomEvent<Select, "change">) {
+		this.fireDecoratorEvent("scope-change", {
+			scope: this.scopeOptions.find(option => e.detail.selectedOption.id === option._id),
```

Comment:
is it possible to introduce some private key to handle that? I dont like using `_id`